src/api/server.rs

```diff
@@ -0,0 +1,123 @@
+use super::{handler, schema::build_schema};
+
+use async_graphql::{
+    http::{playground_source, GraphQLPlaygroundConfig},
+    QueryBuilder,
+};
+use async_graphql_warp::{graphql_subscription, GQLResponse};
+use std::convert::Infallible;
+use std::net::SocketAddr;
+use tokio::sync::oneshot;
+use tokio::sync::oneshot::{Receiver, Sender};
+use warp::filters::BoxedFilter;
+use warp::{http::Response, Filter, Reply};
+
+pub struct Server {
+    /// Address for the API server to bind on
+    address: SocketAddr,
+
+    /// Transmission channel to trigger closure of a running API server
+    trigger_cancel: Sender<()>,
```

Comment:
minor nit/suggestion/question:

I've been getting into the habit of referring to more "exotic" types through their parent module name instead of importing the type itself at the top of the file.

For example, in this case, you'd do `use tokio::sync::oneshot` at the top, and then `oneshot::Sender`. To me, this provides more context at the callsite _what kind of sender_ this is, at the cost of more verbose paths.

Same goes for other types, such as `http::Response` vs `Response`, etc.

There isn't really a hard-and-fast rule we can apply here, but wondering what others think. It's a nit though, so feel free to ignore this comment if you want.

Reply:
👍 I like that, personally. I started to do that with `api::*`, rather than the end structs (although admittedly, not universally.) I do think it helps with readability, and also lessens the import noise.

I'll do a sweep to enforce that more uniformly.